### PR TITLE
Properly support gl_PointSize and gl_PointCoord

### LINKED ISSUE #20979: three.js points billboards example doesn't display anything
https://threejs.org/examples/webgl_points_billboards.html

### LINKED ISSUE #20992: three.js buffergeometry points example renders incorrectly
https://threejs.org/examples/webgl_buffergeometry_points.html
https://threejs.org/examples/webgl_buffergeometry_points_interleaved.html

### LINKED ISSUE #20993: three.js custom attributes points example doesn't render anything
https://threejs.org/examples/webgl_custom_attributes_points.html
https://threejs.org/examples/webgl_custom_attributes_points2.html
https://threejs.org/examples/webgl_custom_attributes_points3.html

### LINKED ISSUE #21007: three.js canvas lights pointlights example renders incorrectly
https://threejs.org/examples/canvas_lights_pointlights.html

### LINKED ISSUE #21719: gl-pointcoord.html doesn't draw anything
The resulting canvas is transparent black in Servo. I haven't been able to isolate the cause of this yet, but it's probably the same problem causing #20993/#20992/#21007/#20979.

### PR DESCRIPTION
**Pull request by nox**
Fixes #21719.
Fixes #20993.
Fixes #20992.
Fixes #21007.
Fixes #20979.

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21961)

### CODE REVIEW

File: components/canvas/canvas_data.rs [hunk 1]
```diff
@@ -99,7 +99,7 @@ impl<'a> CanvasData<'a> {
     }
 
     pub fn fill_text(&self, text: String, x: f64, y: f64, max_width: Option<f64>) {
-        error!("Unimplemented canvas2d.fillText. Values received: {}, {}, {}, {:?}.", text, x, y, max_width);
+        // error!("Unimplemented canvas2d.fillText. Values received: {}, {}, {}, {:?}.", text, x, y, max_width);
```
Comment (jdm) [inline components/canvas/canvas_data.rs]: Revert this, please?

**Review by jdm — APPROVED** [review]

**Comment by nox** [comment]
This didn't fail on macOS, ugh.

**Comment by nox** [comment]
Oh, `try=wpt` only tries on Linux…

**Comment by jdm** [comment]
The answer is OSMesa.

**Comment by nox** [comment]
Interestingly, only `/_webgl/conformance/rendering/point-size.html` is failing, the others are working just fine.

**Comment by jdm** [comment]
Be sure to try individual tests locally with both `--pause-after-test` and `--no-pause-after-test`.

**Comment by nox** [comment]
@jdm Ping?

**Comment by nox** [comment]
https://github.com/servo/gleam/pull/174

**Comment by nox** [comment]
Seems like this test takes a long time to run locally now, and it does use `gl_pointSize`.

@bors-servo try=wpt

**Comment by nox** [comment]
@bors-servo r=jdm

I marked that test as timeout.

**Comment by nox** [comment]
Ugh this put a spurious `tools/wptrunner/wptrunner/update/.wpt-update.lock` entry in the manifest.

**Comment by nox** [comment]
@bors-servo retry

The intermittent failure tracking tool intermittently failed.

**Comment by ferjm** [comment]
@bors-servo retry

- https://github.com/servo/intermittent-failure-tracker/issues/5

**Comment by nox** [comment]
@bors-servo r=jdm

It should work better when I actually commit the expectations.

**Comment by Manishearth** [comment]
So the WebGL cloth example was broken on Oculus Go, and I bisected it to this commit. Any idea what's going on here?

(haven't yet tested it on plain Android, doing so next)

**Comment by Manishearth** [comment]
On a Pixel 2 XL the commit before this (c3c68983) loads some of the cloth thing and then freezes, so I can't test this

**Pull request merged**